cpp/include/raft/neighbors/ivf_pq-inl.cuh

```diff
@@ -156,14 +156,16 @@ void extend(raft::resources const& handle,
  * [n_queries, k]
  * @param[out] distances a device matrix view to the distances to the selected neighbors [n_queries,
  * k]
+ * @param[in] sample_filter a filter the greenlights samples for a given query.
  */
-template <typename T, typename IdxT>
-void search(raft::resources const& handle,
-            const search_params& params,
-            const index<IdxT>& idx,
-            raft::device_matrix_view<const T, IdxT, row_major> queries,
-            raft::device_matrix_view<IdxT, IdxT, row_major> neighbors,
-            raft::device_matrix_view<float, IdxT, row_major> distances)
+template <typename T, typename IdxT, typename SampleFilterT>
+void search_with_filtering(raft::resources const& handle,
+                           const search_params& params,
+                           const index<IdxT>& idx,
+                           raft::device_matrix_view<const T, IdxT, row_major> queries,
+                           raft::device_matrix_view<IdxT, IdxT, row_major> neighbors,
+                           raft::device_matrix_view<float, IdxT, row_major> distances,
```

Comment:
Could you please change the index type of the three arguments spans to `uint32_t` in the new api functions? We were meant to fix this for the existing api as well (https://github.com/rapidsai/raft/issues/1357) but haven't done this yet.

Reply:
@achirkin do you want me to change `template<typename T, typename IdxT, typename SampleFilterT> void search_with_filtering()` to something like `template<typename T, typename IdxT = uint32_t, typename SampleFilterT = detail::NoneSampleFilter> void search_with_filtering()`? I mean I can do it in theory, but it is a separate problem that needs to be addressed separately in a separate PR.
@cjnolet What do you think?

Reply:
The `index_type` template parameter in the `device_matrix_view` is a little misleading because it only applies to the precision used to represent each extent in the underlying mdspan itself and shouldn't really be coupled to the indexing type in the resulting `indices` matrix. Given that this is already a problem that we're planning to address, I'm okay merging these changes as-is for now and cleaning this up when we clean up the others.

Reply:
@alexanderguzhva sorry I didn't clarify. No, as Corey noted I meant to just change the second template parameter in each of these three mdspans to `uint32_t`, because the underlying raw pointer api accepts only 32-bit dimension parameters (`n_queries`, `k`). Don't worry though, I'll add this shortly in a follow-up https://github.com/rapidsai/raft/pull/1539/files#diff-8b0311b97cc4f377506ed342bbe29dc3ac692304f8590feb3397926e8613a658R165-R167